spec/models/match_spec.rb

```diff
@@ -131,4 +131,66 @@
       end
     end
   end
+
+  describe "#multi match fallback" do
+    it "should have other match" do
+      u1 = create(:user,
+        birthdate: Time.now.utc.to_date - 60.years,
+        zipcode: "75001",
+        city: "Paris",
+        geo_citycode: "75001",
+        geo_context: "GEO_CONTEXT")
+      u2 = create(:user,
+        birthdate: Time.now.utc.to_date - 60.years,
+        zipcode: "75001",
+        city: "Paris",
+        geo_citycode: "75001",
+        geo_context: "GEO_CONTEXT")
+
+      vc1 = create(:vaccination_center, :from_paris)
+      vc2 = create(:vaccination_center, :from_paris)
+
+      c1 = create(:campaign, vaccination_center: vc1, ends_at: 2.hours.from_now, available_doses: 1)
+      c2 = create(:campaign, vaccination_center: vc2, ends_at: 2.hours.from_now, available_doses: 1)
+
+      m1 = create(:match, user: u1, campaign: c1, vaccination_center: vc1)
+      m2 = create(:match, :confirmed, user: u2, campaign: c1, vaccination_center: vc1)
+      m3 = create(:match, user: u1, campaign: c2, vaccination_center: vc2)
+
+      m1.set_expiration!
+      m2.set_expiration!
+      m3.set_expiration!
+
+      other = m1.find_other_available_match_for_user
+      expect(other.id).to eq(m3.id)
+    end
+
+    it "should not have other match" do
+      u1 = create(:user,
+        birthdate: Time.now.utc.to_date - 60.years,
+        zipcode: "75001",
+        city: "Paris",
+        geo_citycode: "75001",
+        geo_context: "GEO_CONTEXT")
+      u2 = create(:user,
+        birthdate: Time.now.utc.to_date - 60.years,
+        zipcode: "75001",
+        city: "Paris",
+        geo_citycode: "75001",
+        geo_context: "GEO_CONTEXT")
```

Comment:
Those variable declarations are duplicate with the ones set within the previous test.
You can factor those declarations using `let` outside of the test as follows:
```ruby
let(:users) do
  create_list(:user, 2, birthdate: Time.now.utc.to_date - 60.years,
                        zipcode: "75001",
                        city: "Paris",
                        geo_citycode: "75001",
                        geo_context: "GEO_CONTEXT")
end
let(:u1) { users.first }
let(:u2) { users.second }
```

Reply:
I can't factorize users because I need them linked to a specific number of campaigns